fdbserver/fdbserver.actor.cpp

```diff
@@ -1504,7 +1504,9 @@ struct CLIOptions {
 				maxLogsSize = maxLogs * rollsize;
 			}
 		}
-		machineId = getSharedMemoryMachineId().toString();
+		if (!zoneId.present() && !(localities.isPresent(LocalityData::keyZoneId) && localities.isPresent(LocalityData::keyMachineId))) {
```

Comment:
This condition took me a while to understand. I think the intended effect could be accomplished more simply by replacing occurrences of `machineId` with `getSharedMemoryMachineId().toString()` below. Then we could get rid of `CLIOptions::machineID` altogether. If calling getSharedMemoryMachineId multiple times is problematic, then maybe we could have a local `Optional<std::string>` machineId that we set right before we need it if it's not already set.

Reply:
I don't think we should remove a command line argument in a patch release. I also don't know whether anyone is still using `machineID`... I would like to keep it like this for now and maybe fix it in master.